pkg/costmodel/costmodel.go

```diff
@@ -2293,3 +2293,125 @@ func measureTimeAsync(start time.Time, threshold time.Duration, name string, ch
 		ch <- fmt.Sprintf("%s took %s", name, time.Since(start))
 	}
 }
+
+func (cm *CostModel) QueryAllocation(window kubecost.Window, resolution, step time.Duration, aggregate []string, includeIdle, idleByNode, includeProportionalAssetResourceCosts bool) (*kubecost.AllocationSetRange, error) {
+	// Validate window is legal
+	if window.IsOpen() || window.IsNegative() {
+		return nil, fmt.Errorf("illegal window: %s", window)
+	}
+
+	// Idle is required for proportional asset costs
+	if includeProportionalAssetResourceCosts {
+		includeIdle = true
+	}
```

Comment:
I'm not a fan of silently changing parameters that are incorrectly combined. I'd rather a helpful error be thrown. I can be convinced otherwise, though!

Reply:
I like the idea. A helpful error message to serve as a guardrail. @nikovacevic thoughts?